pandora-client-web/src/editor/components/previewCutter/previewCutter.tsx

```diff
@@ -0,0 +1,221 @@
+		};
+	}, [context]);
+
+	return (
+		<Graphics
+			zIndex={ 0 }
+			interactive={ true }
+			hitArea={ new PIXI.Rectangle(x, y - lineWidth / 2, size, size) }
+			draw={ draw }
+			ref={ graphic }
+			onpointerdown={ onPointerDown }
+			onpointerup={ onPointerUp }
+			onpointerupoutside={ onPointerUp }
+			onpointermove={ onPointerMove }
+		/>
+	);
+}
+
+export function PreviewCutter() {
+	const state = useObservable(PREVIEW_CUTTER);
+	const onChange = React.useCallback((enabled: boolean) => {
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			enabled,
+		};
+	}, []);
+	const setX = React.useCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
+		const x = clamp(parseInt(ev.target.value, 10), -state.size, CharacterSize.WIDTH + state.size);
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			position: {
+				...PREVIEW_CUTTER.value.position,
+				x,
+			},
+		};
+	}, [state.size]);
+	const setY = React.useCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
+		const y = clamp(parseInt(ev.target.value, 10), -state.size, CharacterSize.HEIGHT + state.size);
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			position: {
+				...PREVIEW_CUTTER.value.position,
+				y,
+			},
+		};
+	}, [state.size]);
+	const setSize = React.useCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
+		const size = clamp(parseInt(ev.target.value, 10), PREVIEW_CUTTER_MIN_SIZE, PREVIEW_CUTTER_MAX_SIZE);
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			size,
+		};
+	}, []);
+	const setCentered = React.useCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
+		PREVIEW_CUTTER.value = {
+			...PREVIEW_CUTTER.value,
+			centered: ev.target.checked,
+		};
+	}, []);
+	const createPreviewImage = useEvent(() => {
+		const container = editorSceneContext?.contentRef.current;
+		if (!container) {
+			return;
+		}
+		const exporter = new ImageExporter();
+		exporter.imageCut(
+			container,
+			{
+				x: (state.centered ? ((CharacterSize.WIDTH - state.size) / 2) : state.position.x),
+				y: state.position.y,
+				width: state.size,
+				height: state.size,
+			},
+			'png',
+			{
+				width: PREVIEW_CUTTER_OUTPUT_SIZE,
+				height: PREVIEW_CUTTER_OUTPUT_SIZE,
+			})
+			.then((image) => DownloadAsFile(image, 'preview.png'))
+			.catch((error) => GetLogger('Editor').error('Error exporting image:', error));
+	});
+	return (
+		<FieldsetToggle legend='Preview Cutter' forceOpen={ state.enabled } onChange={ onChange } className='previewCutter'>
+			<div>
+				<label htmlFor='preview-cutter-x'>X</label>
+				<input id='preview-cutter-x' type='number' value={ state.position.x } onChange={ setX } />
+			</div>
+			<div>
+				<label htmlFor='preview-cutter-y'>Y</label>
+				<input id='preview-cutter-y' type='number' value={ state.position.y } onChange={ setY } />
+			</div>
+			<div>
+				<label htmlFor='preview-cutter-size'>Size</label>
+				<input id='preview-cutter-size' type='number' value={ state.size } onChange={ setSize } />
```

Comment:
This one should use either `useRemotelyUpdatedUserInput` or the following suggestion to be more user friendly.
The reason is, that it cannot save all values users might enter, which causes problems when one tries to enter a new value.
For example consider following case:
- Current value is `250`.
- I want to enter `125`, so I select the whole number and start typing `1`
- At this moment the value changes to `50` (`PREVIEW_CUTTER_MIN_SIZE`).
```suggestion
				<input id='preview-cutter-size' type='number' defaultValue={ state.size } onChange={ setSize } />
```

Reply:
added a new `InputNumber` element that should handle this much better